scrape/scrape.go

```diff
@@ -263,7 +263,7 @@ const maxAheadTime = 10 * time.Minute
 
 type labelsMutator func(labels.Labels) labels.Labels
 
-func newScrapePool(cfg *config.ScrapeConfig, app storage.Appendable, jitterSeed uint64, logger log.Logger) (*scrapePool, error) {
+func newScrapePool(cfg *config.ScrapeConfig, app storage.Appendable, jitterSeed uint64, logger log.Logger, reportScrapeTimeout bool) (*scrapePool, error) {
```

Comment:
@roidelapluie if we were to add more metrics here, as suggested in #9285, then having a bool here would mean we'll keep appending more and more bools, maybe it should be some options struct?

Reply:
it should be *a single bool* for all the extra metrics.